nemo/core/classes/modelPT.py

```diff
@@ -152,91 +158,104 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         # ModelPT wrappers over subclass implementations
         self.training_step = model_utils.wrap_training_step(self.training_step)
 
+    def __del__(self):
+        """This is a destructor to delete temporary folders associated with this model instance.
+        """
+        if hasattr(self, "to_delete"):
+            for item in self.to_delete:
+                if os.path.isdir(item):
+                    shutil.rmtree(item)
```

Comment:
Destructors are not guarenteed to be called on clusters since they can terminate the job. Please use only temporary directories and not destructors.

Reply:
got rid of it